swift/ql/lib/codeql/swift/security/XXE.qll

```diff
@@ -0,0 +1,67 @@
+/** Provides classes and predicates to reason about XML external entities (XXE) vulnerabilities. */
+
+import swift
+private import codeql.swift.dataflow.DataFlow
+private import codeql.swift.dataflow.internal.DataFlowPrivate
```

Comment:
What is `DataFlowPrivate` need for?

Reply:
It's for being able to use the `Unit` class `XxeAdditionalTaintStep` extends. Other languages (e.g. Java) expose this class in the public `DataFlow` API, but it's not the case for Swift. Should we fix that instead of using this import?

Reply:
Its a bit weird to use a class from dataflow for this, as it seems to have nothing to do with dataflow.  Seems like ideally it should be defined somewhere outside.  I guess dataflow itself doesn't want to rely on that.

Reply:
Java defines a `Unit.qll` file outside of the dataflow folder, and imports it wherever it's needed. Let me do just that for Swift so that we remove this dependency.

Reply:
https://github.com/github/codeql/pull/11096. I'll rebase when it's merged.

Reply:
This is probably because Java (like many other of our languages) applies the pattern of, for each query, defining a `.qll` file containing (1) abstract classes for the sources, sinks, and sanitizers, and (2) a mechanism for extending the step relation of the configuration (which is the unit class trick we're doing here).

This pattern makes it possible for users to extend the query's behavior without modifying the `.ql` file at all, which is really nice!

This is not a pattern we've applied yet in Swift (since we never settled on whether we were okay with the foot gun related to having dataflow/taintflow configurations in `.qll` files). Now that we have a ql-for-ql query to warn us when this footgun is fired, I think we should consider following the pattern sketched in this PR more consistently for future queries (and possible rewrite our existing queries to follow this. This could be considered tech debt).

Reply:
Sounds good. But I don't want to force my habits from another language if they don't really fit here. Let me know if you want me to move everything back to the `.ql` file as the other queries do.

Reply:
I really like the pattern, so I think it's great to have a template query that we can copy/paste from in the future 😄. It would be good to have @geoffw0's opinion on this as well, though.

Reply:
I don't mind the query in a separate `.qll` pattern, it has advantages for testing and extension.  The disadvantage seems to be a proliferation of files making things a bit harder to navigate, especially when some are in the `src` directory and some in `lib`.  It feels like `lib/codeql/swift/security` could become a mess if we do all taint queries exactly like this one???

Reply:
It would look something like this: https://github.com/github/codeql/tree/main/java/ql/lib/semmle/code/java/security

Reply:
I think I would prefer if it had the same directory structure as `src` and `tests`; though I suppose that would mean `CWE-XYZ` files with only one file (`.qll`) in each directory.  So perhaps I haven't thought this through enough.

Reply:
The bad thing about that is that `CWE-XYZ` directories wouldn't be importable AFAIK.

Reply:
> The bad thing about that is that CWE-XYZ directories wouldn't be importable AFAIK.

Correct :(

Reply:
OK, I'm happy with the way you've arranged this query.  At some point, perhaps after a few weeks of getting used to this way of doing things, we can decide if we want to make the other (dataflow) queries consistent with it (or not).